src/wh_server_crypto.c

```diff
@@ -1963,6 +2007,348 @@ static int _HandleSha256Dma(whServerContext* server, whPacket* packet,
+    wh_Packet_pq_mldsa_verify_Dma64_res* res = &packet->pqMldsaVerifyDma64Res;
+#endif
+
+    /* Transaction state */
+    whKeyId key_id;
+    int     evict = 0;
+
+    if (ctx == NULL || packet == NULL || out_size == NULL) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* Get key ID and evict flag */
+    key_id = WH_MAKE_KEYID(WH_KEYTYPE_CRYPTO, ctx->comm->client_id, req->keyId);
+    evict  = !!(req->options & WH_PACKET_PQ_MLDSAVERIFY_OPTIONS_EVICT);
+
+    /* Initialize key */
+    ret = wc_MlDsaKey_Init(key, NULL, ctx->crypto->devId);
+    if (ret != 0) {
+        return ret;
+    }
+
+    /* Export key from cache */
+    ret = wh_Server_MlDsaKeyCacheExport(ctx, key_id, key);
+    if (ret == 0) {
+        /* Process client signature buffer address */
+        ret = wh_Server_DmaProcessClientAddress(
+            ctx, (uintptr_t)req->sig.addr, &sigAddr, req->sig.sz,
+            WH_DMA_OPER_CLIENT_READ_PRE, (whServerDmaFlags){0});
+
+        if (ret == 0) {
+            /* Process client message buffer address */
+            ret = wh_Server_DmaProcessClientAddress(
+                ctx, (uintptr_t)req->msg.addr, &msgAddr, req->msg.sz,
+                WH_DMA_OPER_CLIENT_READ_PRE, (whServerDmaFlags){0});
+
+            if (ret == 0) {
+                /* Verify the signature */
+                ret = wc_MlDsaKey_Verify(key, sigAddr, req->sig.sz, msgAddr,
+                                         req->msg.sz, &verified);
+
+                if (ret == 0) {
+                    /* Post-read processing of signature buffer */
+                    ret = wh_Server_DmaProcessClientAddress(
+                        ctx, (uintptr_t)req->sig.addr, &sigAddr, req->sig.sz,
+                        WH_DMA_OPER_CLIENT_READ_POST, (whServerDmaFlags){0});
+
+                    if (ret == 0) {
+                        /* Post-read processing of message buffer */
+                        ret = wh_Server_DmaProcessClientAddress(
+                            ctx, (uintptr_t)req->msg.addr, &msgAddr,
+                            req->msg.sz, WH_DMA_OPER_CLIENT_READ_POST,
+                            (whServerDmaFlags){0});
+
+                        if (ret == 0) {
+                            /* Set verification result */
+                            res->verifyResult = verified;
+                            *out_size = WH_PACKET_STUB_SIZE + sizeof(*res);
+                        }
+                    }
+                }
+            }
+        }
+
+        /* Evict key if requested */
+        if (evict) {
+            (void)hsmEvictKey(ctx, key_id);
+        }
+    }
+
+    wc_MlDsaKey_Free(key);
+    return ret;
+#endif /* WOLFSSL_DILITHIUM_NO_VERIFY */
+}
+
+static int _HandleMlDsaCheckPrivKeyDma(whServerContext* server, whPacket* packet,
+                                        uint16_t* size)
+{
+    return WH_ERROR_NOHANDLER;
+}
+#endif /* HAVE_DILITHIUM */
+
+#if defined(HAVE_DILITHIUM) || defined(HAVE_FALCON)
```

Comment:
wc should #define HAVE_PQC as the OR of any of the PQC algos.  No change.  Just a grumble.